Mage.Sets/src/mage/cards/c/CarmenCruelSkymarcher.java

```diff
@@ -0,0 +1,108 @@
+package mage.cards.c;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.effects.common.GainLifeEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldTargetEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.cards.Card;
+import mage.constants.*;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.counters.CounterType;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterPermanentCard;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCardInYourGraveyard;
+
+/**
+ *
+ * @author skaspels
+ */
+public final class CarmenCruelSkymarcher extends CardImpl {
+
+    private static final FilterCard filter = new FilterPermanentCard(
+            " a permanent from your graveyard with mana value less than or equal to {this}'s power");
+    static {
+        filter.add(CarmenPredicate.instance);
```

Comment:
Miss card owner filter (from your graveyard)

Reply:
Doesn't the target class handle that?

Reply:
Yes, fine

Reply:
The ability does hand that - `ability.addTarget(new TargetCardInYourGraveyard(filter));`
will add the min number of targets shortly.